### PR TITLE
fix(blockfrost): handle arrays in image and description fields

### PR DESCRIPTION
**Pull request by jvieiro**
# Context

When using Blockfrost provider, NFT gets rejected with array in image field when parsing a transaction and Array in description field shown as undefined.

# Proposed Solution

Create AssetInfo.TokenMetadata only from response.metadata, since onchain_metadata refers to cip 25.

# Important Changes Introduced

### CODE REVIEW

File: packages/core/src/Asset/types/TokenMetadata.ts
Comment (mkazlauskas) [inline packages/core/src/Asset/types/TokenMetadata.ts:33]: Is this valid? Can't see anything about arrays in the [CIP](https://github.com/cardano-foundation/CIPs/pull/137/files#diff-f478f61a4b67bc7fee1d41bbb427028a7615beaacf726efc9289104e64f26ffaR64)

Comment (jvieiro) [reply packages/core/src/Asset/types/TokenMetadata.ts]: Thanks, I've changed it

File: packages/blockfrost/test/blockfrostAssetProvider.test.ts
Comment (mkazlauskas) [inline packages/blockfrost/test/blockfrostAssetProvider.test.ts:217]: I'm wondering if this is actually the cip25 NFT metadata. If it is then we shouldn't use it as cip38 token metadata :thinking: maybe let's add some e2e test(s) (in `packages/e2e/test/blockfrost`) to fetch metadata of some token and some nft

Comment (mkazlauskas) [reply packages/blockfrost/test/blockfrostAssetProvider.test.ts:217]: So I think the underlying bug is that we're trying to map cip38 (AssetInfo.tokenMetadata) into cip25 (AssetInfo.nftMetadata). We shouldn't be doing that

Comment (jvieiro) [reply packages/blockfrost/test/blockfrostAssetProvider.test.ts:217]: So, `response.onchain_metadata `shouldn't be used to create `AssetInfo.tokenMetadata,` just `response.metadata`?

File: packages/blockfrost/src/blockfrostAssetProvider.ts
Comment (jvieiro) [reply packages/blockfrost/src/blockfrostAssetProvider.ts:11]: It comes from https://github.com/blockfrost/blockfrost-js/blob/master/packages/blockfrost-js/src/types/OpenApi.ts#L4959

File: packages/blockfrost/test/blockfrostAssetProvider.test.ts
Comment (mkazlauskas) [reply packages/blockfrost/test/blockfrostAssetProvider.test.ts:217]: Yes, `response.onchain_metadata` is `AssetInfo.nftMetadata`. We already have a different implementation to get `nftMetadata`, so lets just keep that for now to not break anything.

**Review by mkazlauskas — DISMISSED** [review]
LGTM. Will need to squash into 1 commit after all approvals

**Review by iadmytro — APPROVED** [review]

**Comment by jvieiro** [comment]
> LGTM. Will need to squash into 1 commit after all approvals

Squashed!

**Pull request merged**
